src/rust/engine/options/src/lib.rs

```diff
@@ -158,27 +159,62 @@ pub struct OptionParser {
 
 impl OptionParser {
     pub fn new(env: Env, args: Args) -> Result<OptionParser, String> {
+        let buildroot = BuildRoot::find()?;
+        let buildroot_string = String::from_utf8(buildroot.as_os_str().as_bytes().to_vec())
+            .map_err(|e| {
+                format!(
+                    "Failed to decode build root path {}: {}",
+                    buildroot.display(),
+                    e
+                )
+            })?;
+
+        let mut seed_values = HashMap::from([
+            ("buildroot".to_string(), buildroot_string.clone()),
+            ("homedir".to_string(), shellexpand::tilde("~").into_owned()),
+            ("user".to_string(), whoami::username()),
+        ]);
+        seed_values.extend(
+            env.env
+                .iter()
+                .map(|(k, v)| (format!("env.{k}", k = k), v.clone())),
+        );
+
         let mut sources: BTreeMap<Source, Rc<dyn OptionsSource>> = BTreeMap::new();
         sources.insert(Source::Env, Rc::new(env));
         sources.insert(Source::Flag, Rc::new(args));
         let mut parser = OptionParser {
             sources: sources.clone(),
         };
 
-        let config_path = BuildRoot::find()?.join("pants.toml");
+        fn path_join(a: &str, b: &str) -> String {
```

Comment:
We don't use standard path joining for this because that operates on OsString and we use String here because that's what config deals in.